File: fastAPI-react-app/backend/app/schemas/detection.py
from pydantic import BaseModel
from typing import Dict

class GeneratedFiles(BaseModel):
    original_detected: str
    tampered_detected: str
    difference: str
    threshold: str

class DetectionResponse(BaseModel):
    success: bool
    similarity_score: float
    message: str
    generated_files: GeneratedFiles

    class Config:
        from_attributes = True

File: fastAPI-react-app/frontend/src/types/detection.ts
import React from 'react';

export type DetectionRequest = {
    original_image: File;
    tampered_image: File;
}


export type GeneratedFiles = {
    original_detected: string;
    tampered_detected: string;
    difference: string;
    threshold: string;
}

export type DetectionResponse = {
    success: boolean;
    similarity_score: number;
    message: string;
    generated_files: GeneratedFiles;
}
